test/e2e/e2e.go

```diff
@@ -175,6 +191,64 @@ var _ = Describe("validation", func() {
 			})
 		})
 	})
+
+	Context("Creating AddressPool", func() {
+		It("Should create addresspool CR successfully", func() {
+			By("By creating AddressPool CR")
+			addresspool := &metallbv1alpha.AddressPool{
+				TypeMeta: metav1.TypeMeta{
```

Comment:
I think there's no need to specify this, the go client should be able to infer it from the type

Reply:
kubebuilder example specify it https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/book/src/cronjob-tutorial/testdata/project/controllers/cronjob_controller_test.go  ?

Reply:
Yes, but it's redundant and we never do that. It works even if you remove that part because client-go is able to infer that part from the type of the go object